phoenix-core/src/main/java/org/apache/phoenix/calcite/CalciteUtils.java

```diff
@@ -1036,4 +1048,40 @@ public Void visitCall(RexCall call) {
             return null;
         }
     }
+
+    public static Object convertLiteral(SqlLiteral literal, PhoenixRelImplementor implementor) {
+        try {
+            final SchemaPlus rootSchema = Frameworks.createRootSchema(true);
+            final FrameworkConfig config = Frameworks.newConfigBuilder()
+                    .parserConfig(SqlParser.Config.DEFAULT)
+                    .defaultSchema(rootSchema).build();
+            Planner planner = Frameworks.getPlanner(config);
```

Comment:
I could also abstract this to the SqlOptionNode and pass it in to this method. I had that at one point but ended up opting for this since the code didn't seem like it belonged there. Let me know which you prefer.